### PR TITLE
Build most viewed html within guui

### PR DESCRIPTION
**Pull request by nicl**
(Possible now that there is a relevant API endpoint.)

It looks like the svgs are created dynamically in frontend as part of making assets, but I've hardcoded them for now here.

## What does this change?

Most viewed is now constructed locally rather than using html from the Play frontend.

## Why?

Because templating should be done in GUUI, to learn (for me), and to validate the corresponding PR here: https://github.com/guardian/frontend/pull/19507.

### CODE REVIEW

**Comment by SiAdcock** [comment]
Cooool! 🆒 😎 

Would be nice to apply the styles inline to each element, rather than using `className`. What do you think @nicl?

**Comment by nicl** [comment]
@SiAdcock thanks. Re your question, hmm. I originally just copied and pasted the html (with some tidyup) so I hadn't really considered this.

In some ways I feel like inlining the CSS would clutter things a bit in terms of actually understanding/reading the HTML structure. But I need to try this and see!

Are there advantages to inlining other than making it more explicit where the css belongs/relates to? (I can see that this will help ensure unused css doesn't stick around - which might be the killer argument I guess.)

**Comment by sndrs** [comment]
@nicl you can use `styled` to create preact component with their own style info, I think this is what @SiAdcock was referring to? this keeps the component definition and it's style in the scope of the parent component, but the structure is still keep clear at the point of export. e.g. 

https://github.com/guardian/guui/blob/729545f58d369b1ff31e167f0d75ecab0a215590/src/components/Epic.js#L2-L27

File: src/pasteup/numberSVGs.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        0: (
+            <svg width="34" height="40">
+                <path d="M17 40C7.8 40 0 33.2 0 20.2 0 6.6 8.1 0 17.3 0 26.4 0 34 6.5 34 19.9S26.3 40 17 40zm.1-1c4 0 5.2-6.1 5.2-18.7C22.3 7 21 1.1 17.1 1.1c-4 0-5.4 6-5.4 18.7 0 13 1.4 19.1 5.4 19.2z" />
+            </svg>
+        ),
+        1: (
+            <svg width="21" height="40">
+                <path
+                    d="M0 40h21v-1l-5.4-1.6V0h-4.3L.1 5.4v1.3l5.5-1.4v32.1L0 39.1z"
+                    fillRule="evenodd"
```
Comment (sndrs) [inline src/pasteup/numberSVGs.js]: appreciate this was a copy and paste, but i don't think attribute will actually do anything here?

File: src/components/MostViewed.js [hunk 2]
```diff
@@ -54,16 +55,16 @@ export default class MostViewed extends Component {
     constructor() {
         super();
         this.setState({
-            html: 'Placeholder text',
+            json: { trails: [] },
```
[the remark this answers is not in the document]
Comment (RichieAHB) [reply src/components/MostViewed.js]: Not sure about Preact but calling `setState` may fail in some cases in React (if not now then I’m future) as it’s async so `this.state` may not be available on initial render. I think the preferred is to just set state using `this.state = ...` in the constructor.

**Comment by SiAdcock** [comment]
@nicl sorry my word choice was lazy / clumsy / wrong. @sndrs expressed my thoughts much better (thank you! 💕)

File: src/components/MostViewed.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ import styled from 'preact-emotion';
 import { headline, textEgyptian } from 'pasteup/fonts';
 import palette from 'pasteup/palette';
 import { desktop } from 'pasteup/breakpoints';
+import Numbers from 'pasteup/numberSVGs';
```
[the remark this answers is not in the document]
Comment (nicl) [reply src/components/MostViewed.js]: I wasn't sure how to use them dynamically in this case?

**Comment by nicl** [comment]
@SiAdcock @sndrs are makes more sense! I just copied existing implementation here, but styled is definitely better :)

File: src/components/MostViewed.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sndrs) [reply src/components/MostViewed.js]: i guess you'd have to create your own map of them manually with the imports? maybe a `pasteup/big-numbers/index.js` module that exports them for use, but imports them from individual svgs? it's slightly more long-winded, but it make the svgs themselves viewable and editable in image software

**Comment by nicl** [comment]
@SiAdcock @sndrs okay now fully styled! And I've moved svgs into separate files as suggested.

I have some thoughts on the css/js process which I'll try and write up elsewhere.

File: src/components/MostViewed.js [hunk 4]
```diff
@@ -72,7 +62,29 @@ export default class MostViewed extends Component {
         return (
             <div>
                 <Heading>Most Viewed</Heading>
-                <Main dangerouslySetInnerHTML={{ __html: this.state.html }} />
+                <div>
```
Comment (SiAdcock) [inline src/components/MostViewed.js]: Do we need this wrapper `div`? Or the one beginning on line 69?

Comment (nicl) [reply src/components/MostViewed.js]: Nope! Thanks.

**Comment by nicl** [comment]
@sndrs thanks for feedback - I've removed the redundant markup. :)

**Pull request merged**